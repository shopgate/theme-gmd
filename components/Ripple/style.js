/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

const ripple = css({
  position: 'absolute',
  borderRadius: '50%',
  transformOrigin: '50% 50% 0',
}).toString();

const container = css({
  position: 'absolute',
  zIndex: 0,
  top: 0,
  left: 0,
  width: '100%',
  height: '100%',
}).toString();

export default {
  ripple,
  container,
};
