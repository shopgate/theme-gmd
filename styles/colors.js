/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const colors = (process.env.NODE_ENV !== 'test' && themeConfig && themeConfig.colors) ? themeConfig.colors : {};

export default {
  accent: '#5ccee3',
  accentContrast: '#fff',
  background: '#f8f8f8',
  dark: '#000',
  darkGray: '#eaeaea',
  error: '#ff0000',
  focus: '#fa5400',
  light: '#fff',
  placeholder: '#f2f2f2',
  primary: '#fa5400',
  primaryContrast: '#fff',
  shade3: '#9a9a9a',
  shade4: '#b5b5b5',
  shade5: '#ccc',
  shade6: '#656565',
  shade7: '#eaeaea',
  shade8: '#f7f7f7',
  shade9: '#8d8d8d',
  shade10: '#f4f4f4',
  shade11: '#747474',
  shade12: '#939393',
  success: '#35cc29',
  warning: '#ff9300',
  ...colors,
};
