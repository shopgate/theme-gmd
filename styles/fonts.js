/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { font } = themeConfig;

const styles = {
  family: 'Roboto, Arial, sans-serif',
  rootSize: 16,
  lineHeight: 1.5,
  ...font,
};

css.global('body', {
  font: `${styles.rootSize}px/${styles.lineHeight} ${styles.family}`,
});

export default styles;
