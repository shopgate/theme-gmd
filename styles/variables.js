/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const variables = (process.env.NODE_ENV !== 'test' && themeConfig && themeConfig.variables) ? themeConfig.variables : {};

const materialShadow = 'rgba(0, 0, 0, .117647) 0 1px 6px, rgba(0, 0, 0, .117647) 0 1px 4px';

export default {
  materialShadow,
  gap: {
    xsmall: 4,
    small: 8,
    big: 16,
    bigger: 20,
    xbig: 32,
    xxbig: 64,
  },
  emptyPage: {
    icon: 216,
    titleTopGap: 36,
    buttonVerticalGap: 24,
  },
  navigator: {
    height: 56,
    shadow: materialShadow,
  },
  filterbar: {
    height: 48,
  },
  loadingIndicator: {
    size: 32,
    strokeWidth: 3,
  },
  paymentBar: {
    height: 78,
  },
  ...variables,
};
