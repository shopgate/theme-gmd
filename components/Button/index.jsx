/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BaseButton from '@shopgate/pwa-common/components/Button';
import styles from './style';

const buttonTypes = [
  'plain',
  'regular',
  'primary',
  'secondary',
];

/**
 * The basic button component. This components applies Material Design styles and acts as
 * base to more feature-rich button components such as ActionButton and RippleButton, but can
 * also be used as standalone component or any other kind of new button.
 */
class Button extends Component {
  static propTypes = {
    ...BaseButton.propTypes,
    className: PropTypes.string,
    flat: PropTypes.bool,
    type: PropTypes.oneOf(buttonTypes),
    wrapContent: PropTypes.bool,
  };

  static defaultProps = {
    ...BaseButton.defaultProps,
    className: '',
    flat: false,
    type: 'primary',
    wrapContent: true,
  };

  /**
   * Getter for style depending on prop type.
   * @returns {Object}
   */
  get style() {
    const { flat, disabled } = this.props;

    switch (this.props.type) {
      case 'plain':
        return styles.plain();
      case 'regular':
        return styles.regular(disabled);
      default:
      case 'primary':
        return styles.primary(disabled, flat);
      case 'secondary':
        return styles.secondary(disabled, flat);
    }
  }

  /**
   * Getter for the calculated button props.
   * @returns {Object}
   */
  get buttonProps() {
    const buttonProps = {
      className: this.props.className,
      disabled: this.props.disabled,
      onClick: this.props.onClick,
    };

    return buttonProps;
  }

  /**
   * Renders the component.
   * @return {JSX}
   */
  render() {
    const style = this.style;

    const content = this.props.wrapContent ? (
      <div className={style.content}>
        {this.props.children}
      </div>
    ) : this.props.children;

    return (
      <BaseButton {...this.buttonProps} className={`${style.button} ${this.props.className}`}>
        {content}
      </BaseButton>
    );
  }
}

export default Button;
