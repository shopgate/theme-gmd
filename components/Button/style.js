/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import variables from 'Styles/variables';
import colors from 'Styles/colors';

const buttonPadding = `${variables.gap.small + 1}px ${variables.gap.big}px ${variables.gap.small - 1}px`;

/**
 * Creates the button style.
 * @param {string} text The text color.
 * @param {string|null} background The fill color.
 * @return {Object} The button style.
 */
const button = (text, background) => ({
  position: 'relative',
  display: 'inline-block',
  outline: 0,
  padding: 0,
  textTransform: 'uppercase',
  color: text,
  backgroundColor: background,
  fontWeight: 500,
  minWidth: 64,
  borderRadius: 2,
});

/**
 * The basic content wrapper styles.
 */
const contentWrapper = css({
  padding: buttonPadding,
}).toString();

/**
 * Creates the button styles object.
 * @param {string} textColor The text color.
 * @param {string|null} fillColor The fill color.
 * @return {Object} The styles object.
 */
const createButtonStyles = (textColor, fillColor) => {
  const buttonStyle = {
    ...button(
      textColor,
      fillColor
    ),
  };

  const contentStyle = {
    padding: buttonPadding,
    color: buttonStyle.color,
  };

  return {
    button: css(buttonStyle).toString(),
    content: css(contentStyle).toString(),
  };
};

/**
 * Creates a plain button style object without any styling.
 * @return {Object}
 */
const plain = () => ({
  button: css({
    padding: 0,
    outline: 0,
    border: 0,
    textAlign: 'left',
  }).toString(),
  content: '',
});

/**
 * The regular flat button style.
 * @param {boolean} disabled Whether this button is disabled.
 * @return {Object} An object of style definitions.
 */
const regular = (disabled) => {
  if (disabled) {
    // Flat disabled button style.
    return createButtonStyles(colors.shade4, null);
  }

  // Flat enabled button style.
  return createButtonStyles(colors.dark, null);
};

/**
 * The primary button style.
 * @param {boolean} disabled Whether this button is disabled.
 * @param {boolean} flat Whether this button should be rendered flat.
 * @return {Object} An object of style definitions.
 */
const primary = (disabled, flat) => {
  if (!flat) {
    if (disabled) {
      // Regular disabled button style.
      return createButtonStyles(colors.shade4, colors.shade7);
    }

    // Regular enabled button style.
    return createButtonStyles(colors.accentContrast, colors.accent);
  }

  if (disabled) {
    // Flat disabled button style.
    return createButtonStyles(colors.shade4, null);
  }

  // Flat enabled button style.
  return createButtonStyles(colors.accent, null);
};

/**
 * The secondary button style.
 * @param {boolean} disabled Whether this button is disabled.
 * @param {boolean} flat Whether this button should be rendered flat.
 * @return {Object} An object of style definitions.
 */
const secondary = (disabled, flat) => {
  if (!flat) {
    if (disabled) {
      // Regular disabled button style.
      return createButtonStyles(colors.shade4, colors.shade7);
    }

    // Regular enabled button style.
    return createButtonStyles(colors.primaryContrast, colors.primary);
  }

  if (disabled) {
    // Flat disabled button style.
    return createButtonStyles(colors.shade4, null);
  }

  // Flat enabled button style.
  return createButtonStyles(colors.primary, null);
};

export default {
  plain,
  regular,
  primary,
  secondary,
  contentWrapper,
};
