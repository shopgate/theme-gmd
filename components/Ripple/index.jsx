/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { findDOMNode } from 'react-dom';
import clamp from 'lodash/clamp';
import { shift } from '@shopgate/pwa-common/helpers/data';
import { getOffset } from '@shopgate/pwa-common/helpers/dom';
import colors from 'Styles/colors';
import RippleAnimation from './components/RippleAnimation';
import style from './style';

/**
 * The ripple component.
 */
class Ripple extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    color: PropTypes.string,
    disabled: PropTypes.bool,
    fill: PropTypes.bool,
    onComplete: PropTypes.func,
    overflow: PropTypes.bool,
    size: PropTypes.number,
  };

  static defaultProps = {
    className: '',
    color: colors.dark,
    disabled: false,
    fill: false,
    onComplete: () => {},
    overflow: false,
    size: null,
  };

  /**
   * Constructor.
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);

    this.state = {
      ripples: [],
      nextKey: 0,
      hasRipples: false,
    };

    this.duration = 500;
    this.ignoreNextMouseDown = false;
    this.mounted = false;
    this.offset = null;
    this.position = null;
    this.rootNode = null;
    this.style = null;
  }

  /**
   * Sets the `mounted` flag to true.
   */
  componentDidMount() {
    const { fill } = this.props;

    // Reference to Ripple container (for now and later).
    this.rootNode = findDOMNode(this);

    // Initially set the position values to be in the middle of the element.
    // If the Ripple's starting point has to follow the user's touch
    // Location then this is checked later.
    this.position = {
      x: this.rootNode.offsetWidth / 2,
      y: this.rootNode.offsetHeight / 2,
    };

    // Determine the duration based on the size of the ripple.
    this.duration = clamp(
      Math.round((Math.log(this.rippleSize) * 100) - (fill ? -75 : 50)),
      0,
      10000
    );

    // Set the `mounted` flag to true.
    this.mounted = true;

    // If the Ripple is not allowed to overflow then apply some styles to the parent container.
    this.style = (!this.props.overflow) ? {
      overflow: 'hidden',
      position: 'relative',
    } : null;
  }

  /**
   * Sets the `mounted` flag to false
   */
  componentWillUnmount() {
    this.mounted = false;
  }

  /**
   * Calculates the ripple position from the event.
   * @param {Object} event The event object.
   * @returns {Object} An object containing x and y values for the ripple.
   */
  getRipplePosition(event) {
    // If the Ripple is to fill the element then we should set the
    // Starting position to be where the user clicked within that element.
    if (this.props.fill) {
      // Determine if event is a touch event.
      const isTouchEvent = event.touches && event.touches.length;

      // Find the center points of the event.
      const pageX = isTouchEvent ? event.touches[0].pageX : event.pageX;
      const pageY = isTouchEvent ? event.touches[0].pageY : event.pageY;

      // The pointer location is relative to the element offset.
      this.position.x = pageX - this.offset.left;
      this.position.y = pageY - this.offset.top;
    }

    return this.position;
  }

  /**
   * Calculate the size of the ripple.
   * @returns {number}
   */
  get rippleSize() {
    if (this.props.size) {
      if (this.props.fill) {
        return this.props.size * 2;
      }

      return this.props.size;
    }

    /**
     * We want the Ripple to fill the element. We set the diameter of the Ripple
     * to double the distance of the opposing corners of the node. That way the Ripple
     * will reach the furthest away corner if you click at a corner.
     */
    if (this.props.fill) {
      return Math.sqrt((this.rootNode.offsetWidth ** 2) + (this.rootNode.offsetHeight ** 2)) * 2;
    }

    // Otherwise we set the size to be the smaller of the element's height and width.
    return Math.min(this.rootNode.offsetWidth, this.rootNode.offsetHeight);
  }

  /**
   * Adds a new Ripple to the queue.
   * @param {Object} event The event object.
   * @param {boolean} isTouchGenerated Whether the action was triggered by a touch or click.
   */
  addRipple(event, isTouchGenerated) {
    // If the adding of the Ripple was already triggered
    // By a touch start event, no further action will be done.
    if (this.ignoreNextMouseDown && !isTouchGenerated) {
      this.ignoreNextMouseDown = false;
      return;
    }

    // Get the position of the element and store it.
    this.offset = getOffset(this.rootNode);

    // Receive the x and y position for the new Ripple.
    const { x, y } = this.getRipplePosition(event);

    const ripples = this.state.ripples;

    // Append the new ripple to the ripples array.
    ripples.push(
      <RippleAnimation
        color={this.props.color}
        duration={this.duration}
        fill={this.props.fill}
        key={this.state.nextKey}
        onComplete={this.removeRipple}
        size={this.rippleSize}
        x={x}
        y={y}
      />
    );

    this.ignoreNextMouseDown = isTouchGenerated;

    // Update the state.
    this.setState({
      ripples,
      nextKey: this.state.nextKey + 1,
      hasRipples: true,
    });
  }

  /**
   * Will be triggered from the ripple animation component, when the
   * ripple animation is over. It removes the ripple from the queue again.
   */
  removeRipple = () => {
    if (!this.mounted) {
      return;
    }

    const ripples = shift(this.state.ripples);
    this.props.onComplete();

    this.setState({
      ripples,
      hasRipples: !!(ripples.length),
    });
  };

  /**
   * Triggers adding of a new ripple on touch start event.
   * @param {Object} event The even object.
   */
  handleClick = (event) => {
    if (this.props.disabled) {
      return;
    }
    this.addRipple(event, true);
  };

  /**
   * Renders all the ripples in the queue.
   * @returns {JSX|null}
   */
  renderRipples() {
    // It only needs to render, if there is at least one ripple in the queue.
    if (!this.state.hasRipples) {
      return null;
    }

    return (
      <div className={style.container}>
        {this.state.ripples}
      </div>
    );
  }

  /**
   * Renders the final ripple component including it's contents.
   * @returns {JSX}
   */
  render() {
    return (
      <div
        aria-hidden
        className={this.props.className}
        data-test-id="Ripple"
        onClick={this.handleClick}
        style={this.style}
      >
        {this.renderRipples()}
        {this.props.children}
      </div>
    );
  }
}

export default Ripple;
