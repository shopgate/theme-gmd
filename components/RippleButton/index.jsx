/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Ripple from 'Components/Ripple';
import Button from 'Components/Button';
import style from 'Components/Button/style';

/**
 * The ripple button component is a special derivation of the basic button component
 * that adds a ripple effect when clicked.
 */
class RippleButton extends React.Component {
  static propTypes = {
    ...Button.propTypes,
    rippleClassName: PropTypes.string,
    rippleSize: PropTypes.number,
  };

  static defaultProps = {
    ...Button.defaultProps,
    rippleClassName: '',
    rippleSize: null,
  };

  /**
   * Getter for the calculated button props.
   * @returns {Object}
   */
  get buttonProps() {
    return {
      className: this.props.className,
      disabled: this.props.disabled,
      onClick: this.props.onClick,
      flat: this.props.flat,
      type: this.props.type,
      wrapContent: false,
    };
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    if (this.props.disabled) {
      // Don't show the ripple effect when the button is disabled.
      return (
        <Button {...this.buttonProps} wrapContent>
          {this.props.children}
        </Button>
      );
    }

    const rippleProps = {
      className: `${style.contentWrapper} ${this.props.rippleClassName}`,
      fill: true,
      size: this.props.rippleSize,
    };

    return (
      <Button {...this.buttonProps}>
        <Ripple {...rippleProps}>
          {this.props.children}
        </Ripple>
      </Button>
    );
  }
}

export default RippleButton;
