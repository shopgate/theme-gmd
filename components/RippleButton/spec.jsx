/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import trim from 'lodash/trim';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Ripple from 'Components/Ripple';
import Button from '@shopgate/pwa-common/components/Button';
import styles from 'Components/Button/style';
import RippleButton from './index';

describe('<RippleButton />', () => {
  it('should render as a regular ripple button effect if type is omitted', () => {
    const wrapper = shallow(
      <RippleButton>Press me</RippleButton>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Ripple).render().text()).toEqual('Press me');
  });

  it('should render as a regular ripple button if type is explicitly defined', () => {
    const wrapper = mount(
      <RippleButton type={'regular'}>Press me</RippleButton>
    );

    expect(trim(wrapper.find(Button).props().className)).toEqual(styles.regular(false).button);
    expect(wrapper.find(Ripple).render().text()).toEqual('Press me');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as a primary ripple button', () => {
    const wrapper = mount(
      <RippleButton type={'primary'}>Press me</RippleButton>
    );
    expect(trim(wrapper.find(Button).props().className)).toEqual(styles.primary(false).button);
    expect(wrapper.find(Ripple).render().text()).toEqual('Press me');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as a secondary ripple button', () => {
    const wrapper = mount(
      <RippleButton type={'secondary'}>Press me</RippleButton>
    );
    expect(trim(wrapper.find(Button).props().className)).toEqual(styles.secondary(false).button);
    expect(wrapper.find(Ripple).render().text()).toEqual('Press me');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render as a disabled ripple button', () => {
    const wrapper = mount(
      <RippleButton disabled>Press me</RippleButton>
    );
    expect(wrapper.find(Button).props().disabled).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
