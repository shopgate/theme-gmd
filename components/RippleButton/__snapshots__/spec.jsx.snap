// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RippleButton /> should render as a disabled ripple button 1`] = `
<RippleButton
  className=""
  disabled={true}
  flat={false}
  onClick={null}
  rippleClassName=""
  rippleSize={null}
  type="primary"
  wrapContent={true}
>
  <Button
    className=""
    disabled={true}
    flat={false}
    onClick={null}
    type="primary"
    wrapContent={true}
  >
    <Button
      className="css-5rbc6o "
      disabled={true}
      onClick={null}
    >
      <button
        className="css-5rbc6o  css-1rn65c0"
        data-test-id="Button"
        disabled={true}
        onClick={null}
      >
        <div
          className="css-4uxwj9"
        >
          Press me
        </div>
      </button>
    </Button>
  </Button>
</RippleButton>
`;

exports[`<RippleButton /> should render as a primary ripple button 1`] = `
<RippleButton
  className=""
  disabled={false}
  flat={false}
  onClick={null}
  rippleClassName=""
  rippleSize={null}
  type="primary"
  wrapContent={true}
>
  <Button
    className=""
    disabled={false}
    flat={false}
    onClick={null}
    type="primary"
    wrapContent={false}
  >
    <Button
      className="css-1y1e1q7 "
      disabled={false}
      onClick={null}
    >
      <button
        className="css-1y1e1q7  css-1rn65c0"
        data-test-id="Button"
        disabled={false}
        onClick={null}
      >
        <Ripple
          className="css-8s1fw2 "
          color="#000"
          disabled={false}
          fill={true}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-8s1fw2 "
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            Press me
          </div>
        </Ripple>
      </button>
    </Button>
  </Button>
</RippleButton>
`;

exports[`<RippleButton /> should render as a regular ripple button effect if type is omitted 1`] = `
<Button
  className=""
  disabled={false}
  flat={false}
  onClick={null}
  type="primary"
  wrapContent={false}
>
  <Ripple
    className="css-8s1fw2 "
    color="#000"
    disabled={false}
    fill={true}
    onComplete={[Function]}
    overflow={false}
    size={null}
  >
    Press me
  </Ripple>
</Button>
`;

exports[`<RippleButton /> should render as a regular ripple button if type is explicitly defined 1`] = `
<RippleButton
  className=""
  disabled={false}
  flat={false}
  onClick={null}
  rippleClassName=""
  rippleSize={null}
  type="regular"
  wrapContent={true}
>
  <Button
    className=""
    disabled={false}
    flat={false}
    onClick={null}
    type="regular"
    wrapContent={false}
  >
    <Button
      className="css-6bfwc1 "
      disabled={false}
      onClick={null}
    >
      <button
        className="css-6bfwc1  css-1rn65c0"
        data-test-id="Button"
        disabled={false}
        onClick={null}
      >
        <Ripple
          className="css-8s1fw2 "
          color="#000"
          disabled={false}
          fill={true}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-8s1fw2 "
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            Press me
          </div>
        </Ripple>
      </button>
    </Button>
  </Button>
</RippleButton>
`;

exports[`<RippleButton /> should render as a secondary ripple button 1`] = `
<RippleButton
  className=""
  disabled={false}
  flat={false}
  onClick={null}
  rippleClassName=""
  rippleSize={null}
  type="secondary"
  wrapContent={true}
>
  <Button
    className=""
    disabled={false}
    flat={false}
    onClick={null}
    type="secondary"
    wrapContent={false}
  >
    <Button
      className="css-5xfjjh "
      disabled={false}
      onClick={null}
    >
      <button
        className="css-5xfjjh  css-1rn65c0"
        data-test-id="Button"
        disabled={false}
        onClick={null}
      >
        <Ripple
          className="css-8s1fw2 "
          color="#000"
          disabled={false}
          fill={true}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-8s1fw2 "
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            Press me
          </div>
        </Ripple>
      </button>
    </Button>
  </Button>
</RippleButton>
`;
